### PR TITLE
Added the code for matrix transpose in C++

### PR DESCRIPTION
**Pull request by MayureeS**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description
The code takes input as a matrix and outputs the transpose of given matrix. This code is added to NeoAlgo/C-Plus-Plus/cp

### Checklist

- [Yes] I've read the contribution guidelines.
- [Yes] I've checked the issue list before deciding what to submit.
- [Yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Issue related to this PR : #3421

### CODE REVIEW

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Make README changes as well

**Comment by MayureeS** [comment]
@HarshCasper  Thank you for reviewing my PR. I have updated the README file. Please check if it is correct.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top
- Shift the logic to another function and use main for driver code
- Use proper code Format 
- Remove extra lines of code and spaces
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code

**Comment by MayureeS** [comment]
@HarshCasper @ankitaggarwal23  I have made the changes. But now it is showing 4 failing checks. What should I do now?

File: C-Plus-Plus/README.md [hunk 1]
```diff
@@ -69,6 +69,8 @@
 
 - [All String permutation by using recursion](cp/recursion_string_permutation.cpp)
 
+- [Matrix Transpose](cp/matrix_transpose.cpp)
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/README.md]: Remove this file from cp section and add this code file to math section

File: C-Plus-Plus/cp/matrix_transpose.cpp [hunk 2]
```diff
@@ -0,0 +1,125 @@
+/*
+ * Matrix is a 2-D array.
+ * The transpose of a matrix is simply a flipped version of the original matrix.
+ * We can transpose a matrix by switching its rows with its columns.
+ * This code gives output as transpose of input matrix.
+ * */
+
+#include<iostream>
+using namespace std;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/matrix_transpose.cpp]: Add a new line after using namespace std;

File: C-Plus-Plus/cp/matrix_transpose.cpp [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+void transpose(int frows, int fcolumns, int **finput_matrix, int **ftranspose_matrix)	//Function that computes transpose
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/matrix_transpose.cpp]: shift the comment above the function defination

File: C-Plus-Plus/cp/matrix_transpose.cpp [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    int j = 0;
+    for (i = 0; i < fcolumns; ++i)
+    {
+	for (j = 0; j < frows; ++j)
+	{
+	    //According to definition of transpose, each element of transpose is assigned.
+	    ftranspose_matrix[i][j] = finput_matrix[j][i];
+	}
+    }
+}
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/matrix_transpose.cpp]: Align your code properly give one tab space while writing code inside the body of the statement or condition

File: C-Plus-Plus/cp/matrix_transpose.cpp [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+int main(void)
+{
+    int **input_matrix = NULL;
+    int **transpose_matrix = NULL;
+    int rows = 0;
+    int columns = 0;
+    int i = 0;
+    int j = 0;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/matrix_transpose.cpp]: Change it to like this, it will save your lines

File: C-Plus-Plus/cp/matrix_transpose.cpp [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+    }
+    delete[] input_matrix;
+
+    for (i = 0; i < columns; ++i)
+    {
+        delete[] transpose_matrix[i];
+    }
+    delete[] transpose_matrix;
+    return 0;
+}
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/matrix_transpose.cpp]: Align your code properly give one tab space while writing code inside the body of the statement or condition.

File: C-Plus-Plus/cp/matrix_transpose.cpp [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+ *
+ * Explanation: Here, rows=2 & columns=3. On next line each element of matrix is entered in row wise manner separated by space.
+ * So input matrix is :   1 2 3
+ *                        4 5 6
+ * The code gives output as its transpose.
+ *
+ *
+ * Time Complexity : theta(mn)        where m = number of rows, n = number of columns
+ * Space Complexity: Transpose matrix require extra storage same as input matrix. So space complexity becomes theta(mn)
+ *
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/matrix_transpose.cpp]: Kindly cross check Is it Theta or O(mn)

File: C-Plus-Plus/cp/matrix_transpose.cpp [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+ */
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/matrix_transpose.cpp]: Add a new line at the end of file

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by MayureeS** [comment]
@ankitaggarwal23 I have made the said changes. I hope it is correct now. Really sorry for the inconvenience caused by my silly mistakes.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**